File: src/frontends/paddle/src/op/relu6.cpp
// Copyright (C) 2018-2025 Intel Corporation
// SPDX-License-Identifier: Apache-2.0
//

#include "openvino/frontend/paddle/node_context.hpp"
#include "openvino/frontend/paddle/visibility.hpp"
#include "openvino/opsets/opset6.hpp"

namespace ov {
namespace frontend {
namespace paddle {
namespace op {
NamedOutputs relu6(const NodeContext& node) {
    auto data = node.get_input("X");
    auto threshold = node.get_attribute<float>("threshold", 6.0f);
    return node.default_single_output_mapping({std::make_shared<ov::opset6::Clamp>(data, 0.0, threshold)}, {"Out"});
}

}  // namespace op
}  // namespace paddle
}  // namespace frontend
}  // namespace ov

File: tools/mo/openvino/tools/mo/front/caffe/relu6.py
# Copyright (C) 2018-2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0

from openvino.tools.mo.ops.activation_ops import ReLU6
from openvino.tools.mo.front.extractor import FrontExtractorOp


class ReLU6FrontExtractor(FrontExtractorOp):
    op = 'ReLU6'
    enabled = True

    @classmethod
    def extract(cls, node):
        ReLU6.update_node_stat(node)
        return True
